code/modules/mob/living/carbon/brain/brain_item.dm

```diff
@@ -69,23 +68,36 @@
 	owner.brain_op_stage = 4.0
 
 	var/obj/item/organ/brain/B = src
-	if(istype(B) && istype(owner))
+	if(istype(B) && istype(owner) && owner.internal_organs_by_name[organ_tag] == src)

```

Comment:
I'm having trouble coming up with a better ID checking scheme: Checking for lack of a brainmob would conflict with the scheme I want to use for brainmobs actually dying if you let the brain rot, checking for DNA would only allow 1 level of body-swapping shenanigans. I am unaware of any other means of ID-checking, and which brain is in control of what. @Fox-McCloud